docs/site/Extending-OpenAPI-specification.md

```diff
@@ -0,0 +1,154 @@
+application's specification.
+
+_Read about the extension point/extension patten in
+[documentation](https://loopback.io/doc/en/lb4/Extension-point-and-extensions.html)_
+
+## Adding a New OAS Enhancer
+
+Interface `OASEnhancer` is created in `@loopback/openapi-v3` to describe the
+specification enhancers. A typical OAS enhancer class should have a `name` as
+string, and a function `modifySpec()` to modify the current specification.
+
+For example, to modify the `info` field of an OAS, you can create an
+`InfoSpecEnhancer` that implements interface `OASEnhancer` as follow:
+
+```ts
+import {bind} from '@loopback/core';
+import {
+  defaultMergeFn,
+  asSpecEnhancer,
+  OASEnhancer,
+  OpenApiSpec,
+} from '@loopback/openapi-v3';
+
+/**
+ * A spec enhancer to add OpenAPI info spec
+ */
+@bind(asSpecEnhancer)
+export class InfoSpecEnhancer implements OASEnhancer {
+  // give your enhancer a proper name
+  name = 'info';
+
+  // takes in the current spec, modifies it, and returns a new one
+  modifySpec(spec: OpenApiSpec): OpenApiSpec {
+    const InfoPatchSpec = {
+      info: {title: 'LoopBack Test Application', version: '1.0.1'},
+    };
+    // the example calls a default helper function to merge the fragment spec.
+    const mergedSpec = defaultMergeFn(spec, InfoPatchSpec);
+    return mergedSpec;
+  }
+}
+```
+
+- The class is decorated with a binding template `asSpecEnhancer`.
+- The enhancer has a name as `info`. Name can be used to retrieve a certain
+  enhancer (explained in the
+  [extension point section](#oas-enhancer-service-as-extension-point)).
+- The enhancer changes the current specification's `info` object in function
+  `modifySpec`.
+- It calls [`defaultMergeFn`](#default-merge-function) to merge the
+  specification fragment into the current spec.
+
+### Default Merge Function
+
+Since `modifySpec` has full access to the current spec, it can perform any
+operation: merge, delete, or more complicated changes. This is totally
+determined by the extension contributor.
+
+To apply the basic merging, we provide a default helper function called
+`defaultMergeFn` that leverages
+[`json-merge-patch`](https://github.com/pierreinglebert/json-merge-patch) to
+merge two json objects. You can find its usage in the
+[previous section](#adding-a-new-oas-enhancer)
+
+### Registering an Enhancer
+
+After decorating your enhancer properly with `@bind(asSpecEnhancer)`, you can
+bind it to your application as follow:
+
+```ts
+import {createBindingFromClass} from '@loopback/core';
+import {InfoSpecEnhancer} from './enhancers/infoSpecEnhancer';
+
+class MyApplication extends RestApplication {
+  constructor() {
+    super();
+    this.add(createBindingFromClass(InfoSpecEnhancer));
+  }
+}
+```
+
+## OAS Enhancer Service as Extension Point
+
+The OAS enhancer extension point is created in package `@loopback/openapi-v3`.
+It organizes the registered OAS enhancers, and as the first implementation, it
```

Comment:
What does `as the first implementation` mean?

Reply:
we will have follow-up stories to improve this feature.